app/js/components/ContactForm.jsx

```diff
@@ -25,13 +18,23 @@ class ContactForm extends Component {
         <Form>
           <FormGroup>
             <Label for="nickname">Nickname</Label>
-            <Input type="text" name="nickname" id="nickname" value={nickname} className="form-control"
-                   onChange={(e) => this.changeNickname(e)} validations={[required]}/>
+            <Input type="text"
+                   name="nickname"
+                   id="nickname"
+                   value={username}
+                   className="form-control"
+                   onChange={(e) => this.props.changeUsername(e.target.value)}
+                   validations={[required]}/>
           </FormGroup>
           <FormGroup>
             <Label for="contactCode">Status contact code or Status ENS name</Label>
-            <Input type="text" name="contactCode" id="contactCode" value={contactCode}
-                   className="form-control" onChange={(e) => this.changeContactCode(e)}  validations={[required]}/>
+            <Input type="text"
+                   name="contactCode"
+                   id="contactCode"
+                   value={statusContactCode}
+                   className="form-control"
+                   onChange={(e) => this.props.changeStatusContactCode(e.target.value)}
```

Comment:
Don't you find it annoying to change the redux state for every character? That's why I used a local state and saved on page change. 
I know that it doesn't change much, but I'm not a fan of calling actions 20 times in a row.

Reply:
I don't call redux on each change, `this.props.changeStatusContactCode` is a function coming from the local state of the parent component

Reply:
Omg, sorry. I thought this was the container.